functions/venue.js

```diff
@@ -120,28 +142,22 @@ const createVenueData = (data, context) => {
     }
   }
 
-  switch (data.template) {
-    case VenueTemplate.jazzbar:
-    case VenueTemplate.performancevenue:
-    case VenueTemplate.audience:
-    case VenueTemplate.artpiece:
-    case VenueTemplate.embeddable:
-    case VenueTemplate.firebarrel:
-      venueData.iframeUrl = data.iframeUrl;
-      break;
+  if (IFRAME_TEMPLATES.includes(data.template)) {
+    venueData.iframeUrl = data.iframeUrl;
+  }
+
+  if (ZOOM_URL_TEMPLATES.includes(data.template)) {
+    venueData.zoomUrl = data.zoomUrl;
+  }
```

Comment:
But what was the problem? venueData.iframeUrl was set in the prev implementation as well

Reply:
Here in `exports.createVenueData` it was. The problem was in `exports.updateVenue` where it wasn't included. That's why it worked for create, but not for edit/update.

The additional changes here unify both locations to be 'driven' by the same array, so that there is only a single canonical location to update in future, and thus it will be impossible to make this mistake/forget to update both anymore.

This was also the same issue for a long outstanding `firebarrel` bug, implying that it was a relatively easy mistake to make.

Reply:
see explanation in https://github.com/sparkletown/sparkle/pull/1274#discussion_r595776640